feature-swap-impl/src/main/java/io/novafoundation/nova/feature_swap_impl/domain/swap/LastQuoteStoreProvider.kt

```diff
@@ -0,0 +1,39 @@
+package io.novafoundation.nova.feature_swap_impl.domain.swap
+
+import io.novafoundation.nova.common.data.memory.ComputationalCache
+import io.novafoundation.nova.feature_swap_api.domain.model.SwapQuote
+import io.novafoundation.nova.feature_swap_api.domain.model.SwapQuoteArgs
+import io.novafoundation.nova.runtime.state.SelectedOptionSharedState
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.MutableStateFlow
+
+class LastQuoteStoreSharedStateProvider(
+    private val computationalCache: ComputationalCache,
+) {
+
+    suspend fun create(coroutineScope: CoroutineScope): LastQuoteStoreSharedState {
+        return computationalCache.useCache("LastQuoteStoreSharedState", coroutineScope) {
+            RealLastQuoteStoreSharedState()
+        }
+    }
+}
+
+interface LastQuoteStoreSharedState : SelectedOptionSharedState<Pair<SwapQuoteArgs, SwapQuote>?> {
+
+    fun getLastQuote(): Pair<SwapQuoteArgs, SwapQuote>?
+
+    fun setLastQuote(quote: Pair<SwapQuoteArgs, SwapQuote>?)
+}
+
+class RealLastQuoteStoreSharedState : LastQuoteStoreSharedState {
+
+    override val selectedOption = MutableStateFlow<Pair<SwapQuoteArgs, SwapQuote>?>(null)
+
+    override fun getLastQuote(): Pair<SwapQuoteArgs, SwapQuote>? {
+        return selectedOption.value
+    }
+
+    override fun setLastQuote(quote: Pair<SwapQuoteArgs, SwapQuote>?) {
+        selectedOption.value = quote
+    }
+}
```

Comment:
I propose to use this classes to store last quote state for confirmation screen

Reply:
I think we should either pass everything needed for confirmation using Fragment args or use your approach but only set the state when user goes to continue (not to update on every quote change). Thats beceause we want to fix everything on confirm and do not update it before final confirm validations

Reply:
I think the approach with fragment args is much more readable and understandable. Let me choose the better approach when I start confirmation screen.